src/measurement_kit/cmdline/meek_fronted_requests.cpp

```diff
@@ -0,0 +1,83 @@
+// Part of measurement-kit <https://measurement-kit.github.io/>.
+// Measurement-kit is free software. See AUTHORS and LICENSE for more
+// information on the copying conditions.
+
+#include "../cmdline/cmdline.hpp"
+#include <measurement_kit/nettests.hpp>
+#include <measurement_kit/http.hpp>
+
+#include <iostream>
+
+namespace mk {
+namespace cmdline {
+namespace meek_fronted_requests {
+
+static const char *kv_usage =
+    "usage: measurement_kit meek_fronted_requests [-v] input_file \n";
+
+int main(const char *, int argc, char **argv) {
+    Settings settings;
+    std::string name = argv[0];
+    uint32_t verbosity = 0;
+    std::string expected_body, outer_host, inner_host;
+    mk::nettests::MeekFrontedRequestsTest test;
+    int ch;
+    while ((ch = mkp_getopt(argc, argv, "B:D:H:nv")) != -1) {
+        switch (ch) {
+        case 'B':
+            expected_body = mkp_optarg;
+            break;
+        case 'D':
+            outer_host = mkp_optarg;
+            break;
+        case 'H':
+            inner_host = mkp_optarg;
+            break;
+        case 'n':
+            test.set_options("no_collector", true);
+            break;
+        case 'v':
+            ++verbosity;
+            break;
+        default:
+            std::cout << kv_usage;
+            exit(1);
+        }
+    }
+    argc -= mkp_optind;
+    argv += mkp_optind;
+
+    if ((inner_host.empty() && !outer_host.empty()) ||
+        (outer_host.empty() && !inner_host.empty())) {
+        std::cout << "If you specify one of {outer,inner}_host, "
+                     "you must specify both.\n";
+        std::cout << kv_usage;
+        exit(1);
+    }
+
+    if (inner_host.empty() && outer_host.empty()) {
+        test.runnable->needs_input = true;
```

Comment:
Ah, now I better understand the context. So, basically, look at #1132: the idea is to add there a function called `add_input()` with which you pass input to be processed in addition to input available via file.

With such API in place, it's easier because one is not forced to fiddle with `needs_input`.